core/migration/coordinator.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package migration
+
+import (
+	"context"
+
+	"github.com/juju/description/v4"
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/core/database"
+)
+
+// BaseOperation is a base implementation of the Operation interface.
+// The rollback operation is a no-op by default.
+type BaseOperation struct{}
```

Comment:
This isn't used and doesn't implement the interface it says it does anyway.

Reply:
So it's an abstract operation, more akin to a trait. I'll add the other methods so it returns not implemented.